custom_components/livisi/binary_sensor.py

```diff
@@ -177,3 +199,126 @@ def _handle_coordinator_update(self) -> None:
+        self.off_delay_unique_id: str = device.id + "_duration_number"
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        await super().async_added_to_hass()
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{LIVISI_EVENT}_{self.capability_id}",
+                self.trigger_event,
+            )
+        )
+
+        response = await self.coordinator.aiolivisi.async_get_state(
+            self.capability_id, self.entity_description.key
+        )
+        if response is None:
+            self.update_reachability(False)
+            return
+
+        self.update_reachability(True)
+
+        off_delay = self.get_off_delay()
+        if off_delay is None:
+            self._attr_is_on = False
+            return
+
+        lastChange = response.get("lastChanged")
+        if lastChange is None:
+            self._attr_is_on = False
+            return
+
+        try:
+            lastactive = datetime.fromisoformat(lastChange)
+        except ValueError:
+            # ignore invalid lastChange from SHC
+            self._attr_is_on = False
+            return
+
+        now = datetime.now(timezone.utc)
+        delta = now - lastactive
+
+        self._attr_is_on = delta.seconds < off_delay
+
+        @callback
+        def off_delay_listener(now: Any) -> None:
+            """Switch device off after a delay."""
+            self._delay_listener = None
+            self._attr_is_on = False
+            self.async_write_ha_state()
+
+        remaining_detected_time = off_delay - delta.seconds
+        self._delay_listener = evt.async_call_later(
+            self.hass, remaining_detected_time, off_delay_listener
+        )
+
+    @callback
+    def trigger_event(self, event_data: Any) -> None:
+        """Update the state of the device."""
+        off_delay = self.get_off_delay()
+        if off_delay is None:
+            off_delay = 20.0
+
+        self._attr_is_on = True
+        self.async_write_ha_state()
+
+        # From this example:
+        # https://github.com/home-assistant/core/blob/dev/homeassistant/components/rfxtrx/binary_sensor.py#L21
+        if self._delay_listener:
+            self._delay_listener()
+            self._delay_listener = None
+
+        if self.is_on:
+
+            @callback
+            def off_delay_listener(now: Any) -> None:
+                """Switch device off after a delay."""
+                self._delay_listener = None
+                self._attr_is_on = False
+                self.async_write_ha_state()
+
+            self._delay_listener = evt.async_call_later(
+                self.hass, off_delay, off_delay_listener
+            )
+
+    def get_off_delay(self) -> float:
+        """Get the Delay."""
+        id = self.off_delay_unique_id
+
+        entity_registry = er.async_get(self.hass)
+        entity_id = entity_registry.entities.get_entity_id(
```

Comment:
Using a unique id and retrieving the config item via it (and not via the changeable entity id) should fix the problem...